src/app/components/pages/address-detail/address-detail.component.html

```diff
@@ -1,66 +1,73 @@
-<div class="row">
-  <div class="col-sm-8 -details">
-    <h2>{{ address }}</h2>
-    <p *ngIf="transactions !== undefined">
-      <span># of Transactions</span> {{ transactions.length }}
-    </p>
-    <p *ngIf="balance !== undefined">
-      <span>Balance</span> {{ balance | number:'1.0-6'}}
-    </p>
-  </div>
-  <div class="col-sm-4 -qr-code">
-    <app-qr-code [string]="address" *ngIf="address"></app-qr-code>
+<div class="element-details-wrapper -right-margin">
+  <h2 class="-not-xs">{{ address }}</h2>
+  <h2 class="-xs-only">Address</h2>
+  <div class="element-details">
+    <div class="-row -xs-only">
+      <app-qr-code [string]="address" *ngIf="address"></app-qr-code>
+      <span>Address</span><br class="-xs-only" /> {{ address ? address : loadingMsg }}
+    </div>
+    <div class="-row -tx-number"><span># of Transactions</span><br class="-xs-only" /> {{ transactions ? transactions.length : loadingMsg }} </div>
+    <div class="-row"><span>Balance</span><br class="-xs-only" /> {{ transactions ? ( balance | number:'1.0-6') : loadingMsg }} </div>
   </div>
 </div>
+<div class="-qr-code -not-xs">
+  <app-qr-code [string]="address" *ngIf="address"></app-qr-code>
+</div>
 
-<p *ngIf="transactions === undefined">
+<div class="row -msg-container" *ngIf="!transactions">
```

Comment:
Wouldn't this also trigger if array of transactions is empty?

Reply:
The server does not return empty arrays, in case of not finding transactions a 404 error is returned. However, it is true that it would be better if the code considers the possibility of empty arrays, so I modified it and put the checks back.